diff --git a/node_modules/@astrolicious/i18n/assets/stubs/virtual.mjs b/node_modules/@astrolicious/i18n/assets/stubs/virtual.mjs
index f5cc894..f43fa53 100644
--- a/node_modules/@astrolicious/i18n/assets/stubs/virtual.mjs
+++ b/node_modules/@astrolicious/i18n/assets/stubs/virtual.mjs
@@ -237,13 +237,20 @@ export const switchLocalePath = (locale) => {
 	if (!currentLocaleRoute) {
 		currentLocaleRoute = currentLocaleRoutes
 			.filter((route) => route.params.length > 0)
-			.find(
-				(route) =>
-					JSON.stringify(route.params.sort()) ===
-					JSON.stringify(
-						Object.keys(config.paths.dynamicParams?.[locale] ?? {}).sort(),
-					),
-			);
+			.find((route) => {
+        // Convert the route pattern to a regex pattern
+				let pattern = route.injectedRoute.pattern.replace(/[*.]/g, '\\$&');
+				pattern = Object.keys(
+          config.paths.dynamicParams?.[locale] ?? {}
+        ).reduce(
+          (acc, key) => acc.replace(`[${key}]`, ".*"),
+          pattern,
+        );
+				pattern = pattern.replace(/[-[\]{}()+?,\\^$|#\s]/g, '\\$&');
+				return·new·RegExp(`^${pattern}$`).test(
+          _withoutTrailingSlash(config.paths.pathname)
+        );
+			});
 	}

 	// Fallback
